server/protocols/protocols-smtp/src/test/java/org/apache/james/smtpserver/SMTPServerTest.java

```diff
@@ -244,28 +249,29 @@ protected void initSMTPServer(SMTPTestConfiguration testConfiguration) throws Ex
     }
 
     protected void setUpFakeLoader() throws Exception {
-
-        chain = new MockProtocolHandlerLoader();
-    
-        chain.put("usersrepository", UsersRepository.class, usersRepository);
-    
         dnsServer = new AlterableDNSServer();
-        chain.put("dnsservice", DNSService.class, dnsServer);
-    
-        chain.put("mailStore", MailRepositoryStore.class, mailRepositoryStore);
-
-        chain.put("fileSystem", FileSystem.class, fileSystem);
-
         MemoryRecipientRewriteTable rewriteTable = new MemoryRecipientRewriteTable();
-        chain.put("recipientrewritetable", RecipientRewriteTable.class, rewriteTable);
-
         queueFactory = new MemoryMailQueueFactory(new RawMailQueueItemDecoratorFactory());
         queue = queueFactory.createQueue(MailQueueFactory.SPOOL);
-        chain.put("mailqueuefactory", MailQueueFactory.class, queueFactory);
         MemoryDomainList domainList = new MemoryDomainList(mock(DNSService.class));
         domainList.addDomain(Domain.LOCALHOST);
-        chain.put("domainlist", DomainList.class, domainList);
-        
+
+        chain = MockProtocolHandlerLoader.builder()
+            .put(binder -> binder.bind(DomainList.class).toInstance(domainList))
+            .put(new AbstractModule() {
+                @Provides
+                @Singleton
+                public MailQueueFactory<?> provideActiveMQMailQueueFactory() {
```

Comment:
Why activeMq and not memory ? it's a `MemoryMailQueueFactory`  no ?

Otherwise just for my personal culture, why did you chose to use a module here ? What does it change ?

Reply:
> Why activeMq and not memory ? it's a MemoryMailQueueFactory no ?

I will fix the typo.

> Otherwise just for my personal culture, why did you chose to use a module here ? What does it change ?

The type parameter is anoying, I cannot do a binding like above. I tried using a typeLitteral. It failed. I moved on because it is still way better than before.

Reply:
I succeeded to do it!

```
binder -> binder.bind(new TypeLiteral<MailQueueFactory<?>>() {}).toInstance(queueFactory)
```